python/ray/tests/test_basic.py

```diff
@@ -371,6 +371,56 @@ def foo():
     assert without_options != with_options
 
 
+@pytest.mark.parametrize(
+    "ray_start_cluster_head", [{
+        "num_cpus": 0,
+        "object_store_memory": 75 * 1024 * 1024,
+    }],
+    indirect=True)
+def test_fetch_local(ray_start_cluster_head):
+    cluster = ray_start_cluster_head
+    cluster.add_node(num_cpus=2, object_store_memory=75 * 1024 * 1024)
+
+    @ray.remote
+    class Blocker:
+        def __init__(self):
+            self.block = True
+
+        def is_blocking(self):
+            return self.block
+
+        def unblock(self):
+            self.block = False
+
+    block = Blocker.remote()
+
+    @ray.remote
+    def put():
+        while True:
+            blocking = ray.get(block.is_blocking.remote())
+            if blocking is False:
+                break
+            time.sleep(1)
+        return np.random.rand(5 * 1024 * 1024)  # 40 MB data
+
+    local_ref = ray.put(np.random.rand(5 * 1024 * 1024))
+    remote_ref = put.remote()
+    # Data is not ready in any node
+    (ready_ref, remaining_ref) = ray.wait(
+        [remote_ref], timeout=2, fetch_local=False)
+    assert (0, 1) == (len(ready_ref), len(remaining_ref))
+    ray.wait([block.unblock.remote()])
+
+    # Data is ready in some node, but not local node.
+    (ready_ref, remaining_ref) = ray.wait([remote_ref], fetch_local=False)
+    assert (1, 0) == (len(ready_ref), len(remaining_ref))
+    (ready_ref, remaining_ref) = ray.wait(
+        [remote_ref], timeout=2, fetch_local=True)
+    assert (0, 1) == (len(ready_ref), len(remaining_ref))
+    del local_ref
+    ray.get(remote_ref)
```

Comment:
Can you use `ray.wait(fetch_local=True)` here instead of `ray.get`? To make sure it works once there's enough space.